src/base-app/event.cpp

```diff
@@ -0,0 +1,99 @@
+/*Implemented by Jake*/
+
+#include "event.h"
+
+namespace base{
+    event::event()
+    {
+        mainHost = NULL;
+        hosts = NULL;
+        attendingUsers = NULL;
+        location = NULL;
+    }
+
+    event::event(QString theHost)
+    {
+        setHost(theHost);
+        hosts = NULL;
+        attendingUsers = NULL;
+        location = NULL;
+    }
+
+    base::event::event(QString theHost, QVector<QString> theHosts)
+    {
+        setHost(theHost);
+        setHosts(theHosts);
+        attendingUsers = NULL;
+        location = NULL;
+    }
+
+    event::event(QString theHost, QVector<QString> theHosts, QVector<QString> theUsers)
+    {
+        setHost(theHost);
+        setHosts(theHosts);
+        setUsers(theUsers);
+        location = NULL;
+    }
+
+    event::event(QString theHost, QVector<QString> theHosts, QVector<QString> theUsers, base::venue theVenue)
+    {
+        setHost(theHost);
```

Comment:
You could create a helper function that calls setHost, setHosts, setUsers, setLocation. That would lead to less code duplication and make it easier to change the class later on.